### PR TITLE
Update to newer SDK

### PR DESCRIPTION
**Pull request by wtgodbe**
Part of https://github.com/aspnet/AspNetCore-Internal/issues/3016

Update to a newer 5.0 SDK which includes the `netcoreapp5.0` TFM.

CC @dougbu @JunTaoLuo

### CODE REVIEW

**Comment by JunTaoLuo** [comment]
Need to add NoWarn for NU5131:
```
    <!-- Working around https://github.com/NuGet/Home/issues/8467 -->
    <NoWarn>$(NoWarn);NU5131</NoWarn>
``` 
and NU5048: https://github.com/aspnet/AspNetCore-Internal/issues/3103. This will probably affect several SDK update PRs.

File: src/ProjectTemplates/test/Helpers/TemplatePackageInstaller.cs [hunk 1]
```diff
@@ -90,7 +90,7 @@ private static async Task InstallTemplatePackages(ITestOutputHelper output)
 
             /*
              * The templates are indexed by path, for example:
-              &USERPROFILE%\.templateengine\dotnetcli\v5.0.100-alpha1-013788\packages\nunit3.dotnetnew.template.1.6.1.nupkg
+              &USERPROFILE%\.templateengine\dotnetcli\v5.0.100-alpha1-014696\packages\nunit3.dotnetnew.template.1.6.1.nupkg
```
Comment (dougbu) [inline src/ProjectTemplates/test/Helpers/TemplatePackageInstaller.cs]: Revert changes to this file. Just doesn't matter.

**Review by dougbu — CHANGES_REQUESTED** [review]

**Comment by wtgodbe** [comment]
CI is green - @dougbu @JunTaoLuo does this look good to go? Should I add back the `NetCoreAppMaximumVersion` & `KnownFrameworkReference` workarounds?

**Review by JunTaoLuo — APPROVED** [review]

**Comment by wtgodbe** [comment]
Added the workaround for vNext SDK

**Comment by wtgodbe** [comment]
> eng/Workarounds.targets(37,7): error MSB4113: Specified condition "!(@(KnownAppHostPack->AnyHaveMetadataValue('TargetFramework', '$(DefaultNetCoreTargetFramework)')))" evaluates to "!" instead of a boolean.

Not sure why this isn't working here, the exact same works fine in https://github.com/aspnet/EntityFramework6/pull/1276 & https://github.com/aspnet/EntityFrameworkCore/pull/17945

**Comment by wtgodbe** [comment]
Trying https://github.com/aspnet/AspNetCore/pull/14168/commits/7e5b057ae7fd2791227965d4feed166263636c11, which short-circuits when `KnownFrameworkReference`/`KnownAppHostPack` are empty/don't exist. I'm still not sure why I didn't hit this in the EF repos

File: eng/Workarounds.targets [hunk 2]
```diff
@@ -25,6 +29,20 @@
     <PackageReference Include="Internal.AspNetCore.BuildTasks" PrivateAssets="All" Version="$(InternalAspNetCoreBuildTasksPackageVersion)" IsImplicitlyDefined="true" />
   </ItemGroup>
 
+  <ItemGroup>
+    <!-- Workaround when there is no vNext SDK available, copy known apphost/framework reference info from 3.0 -->
+    <KnownAppHostPack 
+      Include="@(KnownAppHostPack->WithMetadataValue('TargetFramework', 'netcoreapp3.0'))"
+      TargetFramework="$(DefaultNetCoreTargetFramework)"
+      Condition="(@(KnownAppHostPack->Count()) != '0') AND (!(@(KnownAppHostPack->AnyHaveMetadataValue('TargetFramework', '$(DefaultNetCoreTargetFramework)'))))"
```
Comment (dougbu) [inline eng/Workarounds.targets]: Why the extra layer of parentheses around both of the AND'ed clauses?

**Review by dougbu — APPROVED** [review]
Change looks straightforward. Not sure why builds are still failing but know you'll figure it out 😄

File: eng/Workarounds.targets (hunk 2 above)
Comment (wtgodbe) [reply eng/Workarounds.targets]: Paranoia 😆

**Comment by wtgodbe** [comment]
Closing in favor of https://github.com/aspnet/AspNetCore/pull/14206

**Pull request closed without merging**